app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,39 @@
+class CoursesController < ApplicationController
```

Comment:
Okay, Looks good.

And what are supposed to be declared inside <code>module Courses; end</code> ? And for de-coping, I think can define the outside interfaces in it ?